presto-product-tests/src/main/java/com/facebook/presto/tests/sqlserver/Select.java

```diff
@@ -120,24 +119,28 @@ public void testAllDatatypes()
             throws SQLException
     {
         String sql = format(
-                "SELECT bi, si, i, ti, f, r, c, vc, te, nc, nvc, nt, d, dt, dt2, sdt " +
+                "SELECT bi, si, i, ti, f, r, c, vc, te, nc, nvc, nt, d, dt, dt2, sdt, pf30, pf22 " +
                         "FROM %s", ALL_TYPES_TABLE_NAME);
         QueryResult queryResult = onPresto()
                 .executeQuery(sql);
 
         assertThat(queryResult)
                 .hasColumns(BIGINT, SMALLINT, INTEGER, TINYINT, DOUBLE, REAL, CHAR, LONGNVARCHAR, LONGNVARCHAR,
-                        CHAR, LONGNVARCHAR, LONGNVARCHAR, DATE, TIMESTAMP, TIMESTAMP, TIMESTAMP)
+                        CHAR, LONGNVARCHAR, LONGNVARCHAR, DATE, TIMESTAMP, TIMESTAMP, TIMESTAMP, DOUBLE, REAL)
                 .containsOnly(
                         row(Long.MIN_VALUE, Short.MIN_VALUE, Integer.MIN_VALUE, Byte.MIN_VALUE, Double.MIN_VALUE,
-                                Float.MIN_VALUE, "\0   ", "\0", "\0", "\0   ", "\0", "\0",
+                                Float.MIN_VALUE, "\0   ", "\0", "\0", "\0    ", "\0", "\0",
                                 Date.valueOf("0001-01-02"), Timestamp.valueOf("1753-01-01 00:00:00.000"),
-                                Timestamp.valueOf("0001-01-01 00:00:00.000"), Timestamp.valueOf("1900-01-01 00:00:00")),
+                                Timestamp.valueOf("0001-01-01 00:00:00.000"), Timestamp.valueOf("1900-01-01 00:00:00"),
+                                Double.MIN_VALUE, Float.MIN_VALUE
+                        ),
                         row(Long.MAX_VALUE, Short.MAX_VALUE, Integer.MAX_VALUE, Byte.MAX_VALUE, Double.MAX_VALUE,
-                                Float.MAX_VALUE, "abcd", "abcd", "abcd", "abcd", "abcd", "abcd",
+                                Float.MAX_VALUE, "abcd", "abcdef", "abcd", "abcde", "abcdefg", "abcd",
                                 Date.valueOf("9999-12-31"), Timestamp.valueOf("9999-12-31 23:59:59.997"),
-                                Timestamp.valueOf("9999-12-31 23:59:59.999"), Timestamp.valueOf("2079-06-06 00:00:00")),
-                        row(null, null, null, null, null, null, null, null, null, null, null, null)
+                                Timestamp.valueOf("9999-12-31 23:59:59.999"), Timestamp.valueOf("2079-06-06 00:00:00"),
+                                Double.valueOf("1234567891234567"), Float.valueOf("12345678")
```

Comment:
Why don't you use any fraction numbers?
Integers up to 2^24 can be precisely expressed by float (and up to 2^53 for doubles) IIRC.